File: ros2_demo_py/ros2_demo_py/talker.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import Int64, String

class Talker(Node):
    def __init__(self):
        super().__init__("ROS2_Talker")

        self.pub_ = self.create_publisher(String, "message", 10)
        self.timer_ = self.create_timer(0.5, self.publish_msg)

        self.get_logger().info("Hello From ROS2 Talker!!")

    def publish_msg(self):
        msg = String()
        msg.data = "Hello World"
        self.pub_.publish(msg)


def main():
    rclpy.init()
    node = Talker()
    rclpy.spin(node)
    rclpy.shutdown()
    

if __name__ == "__main__":
    main()

File: ros2_demo_cpp/src/talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Talker : public rclcpp::Node
{
public:
    Talker(): Node("talker")
    {
        publisher_ = this->create_publisher<std_msgs::msg::String>("message", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&Talker::publish_msg, this));
        RCLCPP_INFO(this->get_logger(), "Hello From ROS2 Talker!!");
    }

private:
    void publish_msg()
    {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello World";
        publisher_->publish(msg);
    }

    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    
    auto node = std::make_shared<Talker>();
    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}